src/geometry/WorkingGeometry.ts

```diff
@@ -143,22 +148,60 @@ export class WorkingGeometry {
     /**
      * Scale by pulling a point to a new point.
      *
-     * @param {vec3} pullPoint: point that scales to destination_point after transformation.
-     * @param {vec3} destinationPoint: point that is the result of pull_point after transformation.
+     * @param {vec3} pullPoint: point that scales to destinationPoint after transformation.
+     * @param {vec3} destinationPoint: point that is the result of pullPoint after transformation.
      * @param {vec3} holdPoint: point to scale from, default to true origin.
      */
     public scale(pullPoint: vec3, destinationPoint: vec3, holdPoint: vec3 = vec3.create()) {
-        const destinationVector: vec3 = vec3.sub(vec3.create(), destinationPoint, holdPoint);
-        const pullVector: vec3 = vec3.sub(vec3.create(), pullPoint, holdPoint);
-        const scalingVector: vec3 = this.getScalingVector(destinationVector, pullVector);
-        const scalingMatrix: mat4 = mat4.fromRotationTranslationScaleOrigin(
+        const moveHoldToOrigin = mat4.translate(
             mat4.create(),
-            quat.create(),
-            vec3.create(),
-            scalingVector,
-            holdPoint
+            mat4.create(),
+            vec3.negate(vec3.create(), holdPoint)
         );
-        this.transform(scalingMatrix);
+
+        const translatedPullPoint = vec3.sub(vec3.create(), pullPoint, holdPoint);
+        const translatedDestinationPoint = vec3.sub(vec3.create(), destinationPoint, holdPoint);
+
+        const rotPullToXAxis = mat4.fromQuat(
+            mat4.create(),
+            quat.rotationTo(
+                quat.create(),
+                vec3.normalize(vec3.create(), translatedPullPoint),
+                vec3.fromValues(1, 0, 0)
+            )
+        );
+
+        const scalePullToDestOnXAxis = mat4.scale(
+            mat4.create(),
+            mat4.create(),
+            vec3.fromValues(
+                vec3.len(translatedDestinationPoint) / vec3.len(translatedPullPoint),
+                1,
+                1
+            )
+        );
+
+        const rotXAxisToDestAxis = mat4.fromQuat(
+            mat4.create(),
+            quat.rotationTo(
+                quat.create(),
+                vec3.fromValues(1, 0, 0),
+                vec3.normalize(vec3.create(), translatedDestinationPoint)
+            )
+        );
+
+        const moveOriginToHold = mat4.translate(mat4.create(), mat4.create(), holdPoint);
+
+        // Assemble all the matrices!
+        // Note that matrix is modified after each mul call, but the linter can't figure that out,
+        // so the const is only there to appease the linter
+        const matrix = moveOriginToHold;
+        mat4.mul(matrix, matrix, rotXAxisToDestAxis);
+        mat4.mul(matrix, matrix, scalePullToDestOnXAxis);
+        mat4.mul(matrix, matrix, rotPullToXAxis);
+        mat4.mul(matrix, matrix, moveHoldToOrigin);
```

Comment:
I'm having a hard time following what each of these matrices do, and why they're necessary visually.

I was under the impression that the function `fromRotationTranslationScaleOrigin` allowed us to no longer have to specify each matrix and multiply them together (e.g., no need to translate to origin and then rotate). The error you mentioned before I thought could be fixed in `getScalingVector` because it was scaling to 0 instead of 1 in certain cases.

I'd rather we still use `fromRotationTranslationScaleOrigin` in order to have parity with `scaleByFactor` since they're essentially supposed to be doing the same things, and that function actually fits the purpose of `scale` much better than `scaleByFactor` so I think it's possible. This is assuming that it can still be done without witchcraft and black magic 😃

Reply:
So the order of `mat4.mul` multiplies the first operand by the second one, so this block actually needs to be read from the bottom up, unfortunately. This is the way I'm visualizing what this transformation is doing:
1) Apply a translation so that `holdPoint` is the origin.
2) Apply a rotation so that `pullPoint` is on the x-axis.
3) Apply a scale along the x-axis so that `pullPoint` has the same distance from `holdPoint` (which is currently the origin) as `destinationPoint`.
4) Apply a rotation so that the x axis is collinear with the vector from `holdPoint` to `destinationPoint`. This transformation makes the scaled `pullPoint` move to `destinationPoint`.
5) Apply a translation so that `holdPoint` is back where it was at the beginning of the sequence.

The idea is that we're "stretching" the pull point to the destination, and the object we're stretching is flexible. This means that we expect a shear to happen with this transformation, so we do a rotation + non-uniform scale + rotation to simulate that. I think non-uniform scaling is useful to have so we can do things like stretch spheres into ellipsoids, so this function and `scaleByFactor` should do two different things. The rotations also ensure that the pull and destination points don't have to be collinear, which might be flexibility we want to support.